Modules/IO/TIFF/src/itkTIFFImageIO.cxx

```diff
@@ -570,12 +570,12 @@ TIFFImageIO::InternalWrite(const void * buffer)
     pages = m_Dimensions[2];
   }
 
-  int  scomponents = this->GetNumberOfComponents();
-  auto resolution_x = static_cast<float>(m_Spacing[0] != 0.0 ? 25.4 / m_Spacing[0] : 0.0);
-  auto resolution_y = static_cast<float>(m_Spacing[1] != 0.0 ? 25.4 / m_Spacing[1] : 0.0);
+  unsigned int scomponents = this->GetNumberOfComponents();
+  auto         resolution_x = static_cast<double>(m_Spacing[0] != 0.0 ? 25.4 / m_Spacing[0] : 0.0);
+  auto         resolution_y = static_cast<double>(m_Spacing[1] != 0.0 ? 25.4 / m_Spacing[1] : 0.0);
```

Comment:
Please reconsider the original `static_cast<float>` for both resolution_x and resolution_y. They must be passed to `TIFFSetField` as 32-bit float TIFF tag values. Right?

Reply:
yep

Reply:
Thanks for the clarification! So `TIFFSetField` supports 64-bit `double` as argument (32-bit `float` arguments get promoted to `double`: https://stackoverflow.com/questions/11270588/variadic-function-va-arg-doesnt-work-with-float), but then its value is internally casted to 32-bit `float` anyway 😞 as you pointed out.

Reply:
But still I would like to remark that the expression `m_Spacing[0] != 0.0 ? 25.4 / m_Spacing[0] : 0.0` already is of type `double`. So  `static_cast<double>` seems unnecessary. If you really want to make the type explicit, why not just replace `auto` by `double`? You might consider braced initialization, to improve type safety, something like this:

       double resolution_x { m_Spacing[0] != 0.0 ? 25.4 / m_Spacing[0] : 0.0 };